app/assets/javascripts/discourse/app/lib/search.js

```diff
@@ -259,3 +259,40 @@ export function logSearchLinkClick(params) {
     },
   });
 }
+
+/**
+ * Reciprocal Ranking Fusion Algorithm (RRF)
+ * A method used to combine rankings from multiple sources
+ * to aggregate them to provide a single improved ranking
+ *
+ * RRF = 1 / k + r(d)
+ *
+ * k = a constant, small positive value to avoid division by zero
+ * r(d) = the reciprocal rank of the item in the ranking list
+ *
+ * @param {Array} results - an array of search results
+ *
+ * Provide a single result array to assign a reciprocal rank to each result
+ * Then use _sortByReciprocalRanking() to sort the results by their reciprocal rank
+ **/
+export function reciprocalRankingAlgorithm(results) {
+  const k = 5;
+
+  results.forEach((result, index) => {
+    result.reciprocalRank = 1 / (index + k);
+  });
+
+  return results;
+}
+
+/**
+
+ * Sorts an array of mixed search results by their reciprocal rank
+ * Before sorting, use reciprocalRankingAlgorithm() to assign a reciprocal rank to each result
+ *
+ * @param {Array} results - an combined array of search results from multiple sources
+ *
+ **/
+export function sortByReciprocalRanking(results) {
+  return results.sort((a, b) => b.reciprocalRank - a.reciprocalRank);
+}
```

Comment:
Can't we combine those and expose a single API that receives a list of lists and returns a single sorted list ?